MantidQt/MantidWidgets/src/DataProcessorUI/DataProcessorGenerateNotebook.cpp

```diff
@@ -49,23 +47,22 @@ specified via the corresponding hinting line edit in the view
 @returns ipython notebook string
 */
 DataProcessorGenerateNotebook::DataProcessorGenerateNotebook(
-    std::string name, const std::string instrument,
+    QString name, const QString instrument,
```

Comment:
In the original code `std::string` was sometimes passed by value and sometimes by `const&` when making the changes I tried not to change which technique was used as I too wasn't sure if there was a reason for it.

It seems to be a contentious issue [here](https://stackoverflow.com/questions/2104328/whats-the-best-way-to-send-qstrings-in-a-function-call) since QString uses copy-on-write internally.